Odata-docs/webapi-8/getting-started.md

```diff
@@ -0,0 +1,381 @@
+
+## Create an ASP.NET Core application
+In Visual Studio, create a new C# project from the **ASP.NET Core Empty** template. Name the project "Lab01".
+
+In the **Additional information** dialog, select either of **.NET Core 3.1 (Long Term Support)** or **.NET 6.0 (Long Term Support)** frameworks. Uncheck "Configure for HTTPS" checkbox. Click **Create**.
+
+## Add Microsoft.AspNetCore.OData library package
+Using the **Nuget Package Manager** (under **Tools** menu) install the `Microsoft.AspNetCore.OData` nuget package and all the dependencies.
+
+![Screenshot of installing microsoft aspnetcore odata nuget package](../assets/2022-11-11-aspnet-core-odata-8-getting-started-nuget-package.png)
+
+## Define the CLR types
+Add a folder named **Models** to the project and then add the following C# classes.
+
+**`Order` class**
+```csharp
+namespace Lab01.Models
+{
+    public class Order
+    {
+        public int Id { get; set; }
+        public decimal Amount { get; set; }
+    }
+}
+```
+
+**`Customer` class**
+```csharp
+namespace Lab01.Models
+{
+    using System.Collections.Generic;
+
+    public class Customer
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public List<Order> Orders { get; set; }
+    }
+}
+```
+The `Order` and `Customer` classes both contain an `Id` property that is conventionally treated as the key property of the entity. Note also that the `Customer` class has a collection property named `Orders` of type `List<Order>`.
+
+## Add an OData controller
+Add a folder named **Controllers** and then add the following C# class.
+```csharp
+namespace Lab01.Controllers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Lab01.Models;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.OData.Query;
+    using Microsoft.AspNetCore.OData.Routing.Controllers;
+
+    public class CustomersController : ODataController
+    {
+        private static Random random = new Random();
+        private static List<Customer> customers = new List<Customer>(
+            Enumerable.Range(1, 5).Select(idx => new Customer
+            {
+                Id = idx,
+                Name = $"Customer {idx}",
+                Orders = new List<Order>(
+                    Enumerable.Range(1, 2).Select(dx => new Order
+                    {
+                        Id = (idx - 1) * 2 + dx,
+                        Amount = random.Next(1, 9) * 10
+                    }))
+            }));
+
+        [EnableQuery]
+        public ActionResult Get()
+        {
+            return Ok(customers);
+        }
+
+        [EnableQuery]
+        public ActionResult Get([FromRoute] int key)
+        {
+            var item = customers.SingleOrDefault(d => d.Id.Equals(key));
+
+            if (item == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(item);
+        }
+    }
```

Comment:
I wonder whether it would be a good idea to have a full CRUD demo in the getting started (i.e. adding methods for Create, Update and Delete). Or would that be overkill for this section?

Reply:
We had planned a separate article for CRUD operations that's next in priority after this one

Reply:
@habbes I agree with @asanyaga. If we covered `Post`, `Put`, `Patch` and `Delete` in this tutorial, it'd become too big since we'd have to provide sample requests as well. A separate article would be a better idea